ED-2069: Add OCI oss support in Deploy/DataPipeline/FlinkPipelineJobs

ddevadat opened the pull request:
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

ddevadat committed:
added artifact upload role for oci oss

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
overwrite file in oss if the file exists

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
added artifact download for oci role

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
added role to install oci-cli

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
added csp choice based cloud cli installation

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
added env variables for oci cli for analytics user

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
install oci cli as analytics user

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
adding oci cli in PATH variable for analytics user

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
install oci-cli in analytics home location

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
disabled cloud cli temporarlily

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
setting env variable with double quotes

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
placed oci cli env variables in /etc/environment file

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
added jinja2 template for oci cli config

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
renamed dir templates to template

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
disabled only oci cli temporarily

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
testing idempotency for oci cli

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

heungheung committed:
change Spark provision task with Ruby 2.6

In release 4.6.0, Ruby 2.6 is being used. Don't know why in 4.10 / 5.0, Ruby suddenly became 2.2.

In fact, there is NO ruby2.2-dev for bionic. Changing to Ruby 2.6 to proceed

ddevadat committed:
added apiversion selector

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
Merge branch 'oci-5.1.0' of https://github.com/ocisunbird/sunbird-data-pipeline into oci-5.1.0

ddevadat committed:
added the oci oss bucket upload

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
added full path of oci from analaytics home

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
correction in oci cli location

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
check bucket existence before creating

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
testing the logic for register variable

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
using full path for oci cli

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
testing the bucket check

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
added the always tag individually

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
place an oci os upload command for every upload

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
corrected typo for oci cli command

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
corrected typo for oci cli command

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
testing secor changes for oci oss

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
added oci-bv as the storage class

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
added storageclass selection

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
added a condition to include secor.azure.properties

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
using S3UploadManager in common.properties

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
update flink-conf for telemetry-extractor

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
hardcoding base.url

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
hardcode base.url with double quotes

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
updated flink-conf for telemetry extractor

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
removed changes from flink-conf

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
temporary change

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
temporary change

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
temporary change

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
added hard code value for region

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
removed the hardcode region for flink

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
added a trailing /

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
removed trailing slash

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
changed base.url for s3

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
hardcode base.url

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
debug level only for hadoop

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
using s3a

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
using s3 url

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
removed hadoop logging

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
testing with sse-c values

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
corrected the typo

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
hardcoded endpoint

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
hard coded endpoint url

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
endpoint as variable

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
removed hadoop specific logging

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
added explicit region

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
removed region flag

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
changed oci specific end point for s3

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
removed customization from flink-conf.yaml

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

heungheung committed:
Merge pull request #2 from ocisunbird/oci-5.1.0-secor

OCI specific changes for Secor

heungheung committed:
Merge pull request #1 from ocisunbird/oci-5.1.0-dp-flinkpipeline-jobs

OCI specific changes for DP FlinkJobs

subhashchandrab committed:
Added oci instance prinicipal env var

subhashchandrab committed:
Added oci instance prinicipal env var

ddevadat committed:
Merge pull request #3 from ocisunbird/oci-5.1.0-JAPACCOE-2924

Added oci instance prinicipal env var

ddevadat committed:
added conditional selection of pg db name

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
update the condtion check

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
changed the condition

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
Merge remote-tracking branch 'upstream/release-5.1.1' into oci-5.1.0-github-99-deploy-job-PR

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
added the varible for sslmode for postgres

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
Merge pull request #9 from ocisunbird/oci-5.1.0-github-99-deploy-job-PR

Oci 5.1.0 GitHub 99 deploy job pr

anandp504 commented on kubernetes/helm_charts/secor/config/secor.common.properties:
```diff
@@ -357,7 +357,8 @@ secor.max.message.size.bytes=100000
 # Class that will manage uploads. Default is to use the hadoop
 # interface to S3.
-secor.upload.manager.class=com.pinterest.secor.uploader.AzureUploadManager
+# secor.upload.manager.class=com.pinterest.secor.uploader.AzureUploadManager
```
@ddevadat Don't we have to use CSP specific upload manager? It looks like you have changed it from AzureUploadManager to S3UploadManager.

ddevadat replied:
secor doesnt have an oci specific upload manager. OCI object storage is s3 api compliant. This means OCI can use S3UploadManager

anandp504 commented on kubernetes/helm_charts/secor/config/secor.common.properties:
```diff
@@ -23,12 +23,12 @@ secor.kafka.topic_blacklist=
 # Leave empty if using IAM role-based authentication with s3a filesystem.
-aws.access.key=
-aws.secret.key=
+aws.access.key={{ $.Values.s3_access_key }}
```
@ddevadat  The secor.common.properties should be left as-is and we should add secor.s3.properties to accommodate s3-like object stores. Please refer to secor.azure.properties file and also secor.partition.properties file where secor.azure.properties is imported.

ddevadat replied:
its restructured now

anandp504 commented on kubernetes/helm_charts/secor/config/secor.partition.properties:
```diff
@@ -23,7 +25,7 @@ secor.kafka.group={{ get (get $.Values.secor_jobs $.Release.Name) "consumer_grou
 
 # S3 path where sequence files are stored.
-secor.s3.path=
+secor.s3.path={{- get (get $.Values.secor_jobs $.Release.Name) "base_path" }}
```
@ddevadat secor.s3.path also needs to be moved to secor.s3.properties file and configured in the s3 related config file.

ddevadat replied:
its restructured now

anandp504 reviewed (changes requested):
@ddevadat Please resolve the PR comments.

ddevadat committed:
changing the document overlay structure

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat committed:
Merge pull request #10 from ocisunbird/oci-5.1.0-github-99-deploy-job-s3

changing the document overlay structure

ddevadat committed:
removed virtualenv_python version

Ubuntu 22.04 is coming with default python 3.10 version and virtual environment creation is always failing when explicitly virtualenv_python value to pip task in ansible. As per the documentation of ansible(https://docs.ansible.com/ansible/latest/collections/ansible/builtin/pip_module.html),
virtualenv_python : The Python executable used for creating the virtual environment. For example python3.5, python2.7. When not specified, the Python version used to run the ansible module is used.

Signed-off-by: Deepak Devadathan <deepak.devadathan@oracle.com>

ddevadat commented on ansible/roles/portal-dashboard/tasks/main.yml:
This change is done to accomodate PR https://github.com/Sunbird-Obsrv/sunbird-data-pipeline/pull/1615

 Deploy/DataPipeline/AdhocScripts fails on ubuntu 22.04

Ubuntu 22.04 is coming with default python 3.10 version and virtual environment creation is always failing when explicitly virtualenv_python value to pip task in ansible. As per the documentation of ansible(https://docs.ansible.com/ansible/latest/collections/ansible/builtin/pip_module.html),
virtualenv_python : The Python executable used for creating the virtual environment. For example python3.5, python2.7. When not specified, the Python version used to run the ansible module is used.
